src/common/task.h

```diff
@@ -29,7 +33,24 @@ class TaskExecutionSpec {
   /// Get the task spec size.
   ///
   /// @return The size of the immutable task spec.
-  int64_t SpecSize();
+  int64_t SpecSize() const;
+
+  /// Get the task's spillback count.
+  ///
+  /// @return The spillback count for this task.
+  int getSpillbackCount() const;
```

Comment:
The naming convention here seems inconsistent to me, but I can never remember the convention. @pcmoritz can you comment?

Reply:
yes. I propose we consider the style I'm using here -- it's more conventional for OO programming and class API, namely lower case action followed by camel-case qualifier for the action. If there's agreement I can make the rest of the TaskExecutionSpec class consistent with this style.

Reply:
Let's not reinvent the wheel here and follow what others do. Getters and setters should be named like variables, see https://github.com/apache/arrow/blob/master/cpp/src/arrow/array.h#L224

This is what Arrow does, what TensorFlow does [1], and even the C++ standard does it that way. It's also mentioned in the Google style guide.

[1] https://github.com/tensorflow/tensorflow/blob/master/tensorflow/cc/framework/ops.h#L42

Reply:
I didn't "re-invent the wheel", see [2] for standard OO naming conventions. Note that in BOTH naming schemes, methods are named as "first word all lower case, subsequent words starting with upper case", even though naming scheme 2 goes with underscores for variable names!

[2] https://atomicobject.com/resources/oo-programming/naming-conventions

Reply:
looking at Tensorflow, camel-cased method names:
[3] https://github.com/tensorflow/tensorflow/blob/master/tensorflow/cc/framework/scope.h

Reply:
Sorry, but all projects we share code with (arrow, plasma, tensorflow) doing it the other way is enough reason to name getters and setters like variables. We need to be consistent.

There is good reason to do it that way. Getters and setters are cheap and cannot fail, so they behave a lot like variables.

The naming convention for functions and methods is PascalCase btw, see https://google.github.io/styleguide/cppguide.html.

Reply:
Just to be clear, I'm not saying methods should be named like variables. Just getters and setters that return just a variable or just set a variable.

If a method is expensive and also happens to return a value, the GetSomething namestyle is fine.

Reply:
ok, I buy your getters and setters naming argument. BTW, there's never a good reason to follow the mistakes of others. Underscores in names is a terrible idea. I understand that it's hard to change that now... 
So, if I were to match the contextual naming style, here's what I'm coming up with -- let me know if this is what you want: 
SpillbackCount(), LastTimeStamp(), SetLastTimeStamp, IncrementSpillbackCount() or do you want this:
spillback_count, last_timestamp() for the getters and SetLastTimeStamp and IncrementSpillbackCount() pascal style for the others?

Reply:
People are not super consistent about this but it seems to me that in this case the canonical naming would be:

spillback_count, last_timestamp, set_last_time_stamp (see https://github.com/apache/arrow/blob/42fc57be9c768add32b278adbee2f6f30b10005d/cpp/src/arrow/ipc/writer.h#L83 and others), and IncrementSpillbackCount

I agree that this is not the best/most consistent way of doing things. Let's just follow Arrow/TensorFlow/the Google style guide wherever possible, what do you think?

Reply:
@pcmoritz , how about I just make it consistent with the class' current style, and we can change the whole class style in a separate PR. This is what I've done and pushed. Please feel free to take another look.